src/main/java/sdk/sample/common/CommonSdk.java

```diff
@@ -0,0 +1,321 @@
+                {
+                    case "NetAppAccountInner":
+                        NetAppAccountInner account = anfClient.getAccounts().getByResourceGroup(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId));
+                        if (account == null)
+                            return;
+
+                        continue;
+
+                    case "CapacityPoolInner":
+                        CapacityPoolInner pool = anfClient.getPools().get(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId),
+                                ResourceUriUtils.getAnfCapacityPool(resourceId));
+                        if (pool == null)
+                            return;
+
+                        continue;
+
+                    case "VolumeInner":
+                        VolumeInner volume = anfClient.getVolumes().get(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId),
+                                ResourceUriUtils.getAnfCapacityPool(resourceId),
+                                ResourceUriUtils.getAnfVolume(resourceId));
+                        if (volume == null)
+                            return;
+
+                        continue;
+
+                    case "SnapshotInner":
+                        SnapshotInner snapshot = anfClient.getSnapshots().get(ResourceUriUtils.getResourceGroup(resourceId),
+                                ResourceUriUtils.getAnfAccount(resourceId),
+                                ResourceUriUtils.getAnfCapacityPool(resourceId),
+                                ResourceUriUtils.getAnfVolume(resourceId),
+                                ResourceUriUtils.getAnfSnapshot(resourceId));
+                        if (snapshot == null)
+                            return;
+                }
+            }
+            catch (Exception e)
+            {
+                Utils.writeWarningMessage(e.getMessage());
+                break;
+            }
+        }
+    }
+
+    public static void waitForVolumeSuccess(NetAppManagementClient anfClient, String resourceGroupName, String accountName, String poolName, String volumeName)
+    {
+        waitForVolumeSuccess(anfClient, resourceGroupName, accountName, poolName, volumeName, 10, 60);
+    }
+
+    public static void waitForVolumeSuccess(NetAppManagementClient anfClient, String resourceGroupName, String accountName, String poolName, String volumeName, int intervalInSec, int retries)
+    {
+        for (int i = 0; i < retries; i++) {
+            try {
+                VolumeInner volume = anfClient.getVolumes().get(resourceGroupName, accountName, poolName, volumeName);
+                if (volume.provisioningState().equalsIgnoreCase("Succeeded"))
+                    break;
+                Utils.threadSleep(intervalInSec * 1000);
```

Comment:
we should be needing the threadsleep with the retry mechanism.

Reply:
Not sure what you mean here

Reply:
don't worry about this one, I saw you are just having the wait inside the try block. you should be good on this.